File: server_client_C_sharp/mpp_c_app1/FestivalPersistence/DBRepoPersoana.cs
using System;
using System.Collections.Generic;
using log4net;
using System.Data.SQLite;
using FestivalModel;
using SQLiteCommand = System.Data.SQLite.SQLiteCommand;
using SQLiteConnection = System.Data.SQLite.SQLiteConnection;


namespace FestivalPersistence
{

    public interface IDBRepoPersoana : IRepository<long, Persoana>
    {
    }

    public class DBRepoPersoana : IDBRepoPersoana
    {
        private SQLiteConnection connection;
        private static readonly ILog logger = LogManager.GetLogger("DBRepoPersoana");

        public DBRepoPersoana(string connectionString)
        {
            logger.Info("Initializing DBRepoPersoana...");
            connection = new SQLiteConnection(connectionString);
        }


        public IEnumerable<Persoana> findAll()
        {
            logger.InfoFormat("FindAll method called.");
            List<Persoana> persoane = new List<Persoana>();
            using (SQLiteCommand command = new SQLiteCommand("SELECT * FROM persoana", connection))
            {
                connection.Open();
                using (SQLiteDataReader reader = command.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        int id = reader.GetInt32(0);
                        String nume = reader.GetString(1);
                        String prenume = reader.GetString(2);

                        Persoana p = new Persoana(id, nume, prenume);
                        persoane.Add(p);
                    }
                }

                connection.Close();
            }

            logger.InfoFormat("Found {0} persoane", persoane.Count);
            return persoane;
        }

        public Persoana findOne(long id)
        {
            logger.InfoFormat("findOne method called for id: {0}", id);
            using (SQLiteCommand command =
                   new SQLiteCommand("SELECT * FROM persoana WHERE IDPerosana = @id", connection))
            {
                command.Parameters.AddWithValue("@id", id);
                connection.Open();
                using (SQLiteDataReader reader = command.ExecuteReader())
                {
                    if (reader.Read())
                    {
                        int persoanaId = reader.GetInt32(0);
                        string nume = reader.GetString(1);
                        string prenume = reader.GetString(2);
                        connection.Close();
                        return new Persoana(persoanaId, nume, prenume);
                    }
                }

                connection.Close();
            }

            return null;
        }


        public void save(Persoana entity)
        {
            logger.InfoFormat("save method called for persoana with id: {0}", entity.Id);
            using (SQLiteCommand command =
                   new SQLiteCommand("INSERT INTO persoana (nume, prenume) VALUES ( @nume, @prenume)", connection))
            {
                //command.Parameters.AddWithValue("@id", entity.Id);
                command.Parameters.AddWithValue("@nume", entity.Nume);
                command.Parameters.AddWithValue("@prenume", entity.Prenume);
                connection.Open();
                command.ExecuteNonQuery();
                connection.Close();
            }
        }

        public void delete(long id)
        {
            logger.InfoFormat("delete method called for id: {0}", id);
            using (SQLiteCommand command = new SQLiteCommand("DELETE FROM persoana WHERE id = @id", connection))
            {
                command.Parameters.AddWithValue("@id", id);
                connection.Open();
                command.ExecuteNonQuery();
                connection.Close();
            }
        }
    }
}

File: server_C_sharp_client_Java_final_protoV3/JavaSocketsFestival-CLIENT_java/Java11ChatSockets/ChatPersistence/src/main/java/festival/persistence/repository/DBRepoPersoana.java
package festival.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import festival.model.Artist;
import festival.model.Identifier;
import festival.model.Persoana;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.Properties;

public class DBRepoPersoana  implements IDBRepoPersoana{
    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public DBRepoPersoana(Properties props) {
        logger.info("Initializing DBRepoPersoana with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }



    @Override
    public Iterable findAll() {

        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Persoana> persoane = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from \"persoana\"")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id_persoana = result.getInt("IDPerosana");
                    String nume=result.getString("nume");
                    String prenume=result.getString("prenume");
                    Persoana persoana= new Persoana(nume,prenume);
                    persoana.setId((long) id_persoana);

                    persoane.add(persoana);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(persoane);
        return persoane;

    }

    @Override
    public Optional save(Identifier entity) {
        logger.traceEntry("saving task {} ", entity);
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("insert into persoana (nume,prenume) values (?,?)")) {
            preStmt.setString(1, ((Persoana)entity).getNume());
            preStmt.setString(2,  ((Persoana)entity).getPrenume());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return Optional.ofNullable(entity);
    }

    @Override
    public Optional findOne(Object longID) {

        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Persoana> persoane = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from \"persoana\" where IDPerosana=?")) {
            preStmt.setInt(  1, (Integer) longID);
            try (ResultSet result = preStmt.executeQuery()) {

                    int id_persoana = result.getInt("IDPerosana");
                    String nume=result.getString("nume");
                    String prenume=result.getString("prenume");
                    Persoana persoana= new Persoana(nume,prenume);
                    persoana.setId((long) id_persoana);
                     return Optional.ofNullable(persoana);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(persoane);
       return Optional.empty();

    }

    @Override
    public Optional delete(Object aLong) {
        return Optional.empty();
    }



}
